### PR TITLE
Add remove button

### PR DESCRIPTION
**Pull request by Laylo309**
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button (use filter() method).
- All changes to the To-Do List be saved in local storage.

### CODE REVIEW

File: src/Assets/CRUD.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    if (index > -1) {
+      const newData = this.getAllTasks();
+      newData.splice(index, 1);
+      for (let i = index; i < this.getAllTasks().length - 1; i += 1) {
+        newData[i].index = String(i, 10);
+      }
+      this.updateTask(newData);
+      this.showTask();
+    }
+  }
```
Comment (abruzy) [inline src/Assets/CRUD.js:41]: - [ ] `delete` functionality works well but when I have only one item and I try to delete it or edit it, it doesn't work
Screenshot 2021-08-21 at 21 57 23

**Review by abruzy — CHANGES_REQUESTED** [review]
Hi @Laylooo 👋🏽, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- implementing onKey press to add the task to the list of todos.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Assets/CRUD.js (hunk 1 above)
Comment (Laylooo) [reply src/Assets/CRUD.js:41]: In my case, everything working well. All of my coding partners tried but in their browser, they didn't have any problem.

Comment (abruzy) [reply src/Assets/CRUD.js:41]: this is resolved now 🦾

Comment (abruzy) [inline src/Assets/CRUD.js:41]: Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
Screenshot 2021-08-21 at 22 24 28

**Review by abruzy — CHANGES_REQUESTED** [review]
Hi @Laylooo 👋🏽, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Assets/CRUD.js (hunk 1 above)
Comment (Laylooo) [reply src/Assets/CRUD.js:41]: image
Indexes also working well

Comment (Laylooo) [reply src/Assets/CRUD.js:41]: I cannot understand why are those things happening in your case but when I checked everything in my browser it is working properly.

**Review by descholar-ceo — APPROVED** [review]
Hi @Laylooo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have only 3 code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by SpaYco — COMMENTED** [review]
Hello 👋,

_This is an additional review as a result of the second opinion request._

It looks like everything got resolved here, so I have nothing else to say other than good luck with your future projects!

Keep rocking 🚀!

**Pull request merged**